packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrixAndCombine_def.hpp

```diff
@@ -172,6 +173,99 @@ unpackRow(const typename PackTraits<GO>::output_array_type& gids_out,
   return 0; // no errors
 }
 
+/// \brief Unpack a single row, in parallel, of a CrsMatrix
+///
+/// \tparam ST The type of the numerical entries of the matrix.
+///   (You can use real-valued or complex-valued types here, unlike
+///   in Epetra, where the scalar type is always \c double.)
+/// \tparam LO The type of local indices.  See the
+///   documentation of Map for requirements.
+/// \tparam GO The type of global indices.  See the
+///   documentation of Map for requirements.
+template<class ST, class LO, class GO, class ExecutionSpace>
+KOKKOS_FUNCTION int
```

Comment:
Why do you use KOKKOS_FUNCTION instead of KOKKOS_INLINE_FUNCTION ? This is defined in a header file. If I remember correctly, KOKKOS_FUNCTION is meant for the case that we use relocatable device code (a device code is separately compiled in a separate object file from kernel launching code). @crtrott Could you make it clear about the kokkos_function usage ? I might be wrong.

Reply:
my mistake,